src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py

```diff
@@ -188,941 +187,1485 @@
+        args['analysis_year'] (int): the year of the final analysis.
+        args['do_economic_analysis'] (bool): Whether to do valuation.
+        args['use_price_table'] (bool): Whether to use a table of annual carbon
+            prices for valuation.  Defaults to ``False``.
+        args['price_table_path'] (string): The path to a table of prices to use
+            for valuation.  Required if ``args['use_price_table']`` is
+            ``True``.
+        args['inflation_rate'] (number): The rate of inflation.  The number
+            provided is multiplied by ``0.01`` to compute the actual rate of
+            inflation.  Required if ``args['use_price_table']`` is ``False``.
+        args['price'] (number): The carbon price.  Required if
+            ``args['use_price_table']`` is ``False``.
+        args['discount_rate'] (number): The discount rate.  The number provided
+            is multiplied by ``0.01`` to compute the actual discount rate.
+            Required if ``args['do_economic_analysis']``.
+        args['biophysical_table_path'] (string): The path to the biophysical
+            table on disk.  This table has many required columns.  See
+            ``ARGS_SPEC`` for the required columns.
+        args['landcover_transitions_table'] (string): The path to the landcover
+            transitions table, indicating the behavior of carbon when the
+            landscape undergoes a transition.
 
     Returns:
-        transition_idx (int): the current transition
+        ``None``.
+
     """
-    for i in range(0, transitions):
-        if timestep < (snapshot_years[i+1] - snapshot_years[0]):
-            return i
+    suffix = utils.make_suffix_string(args, 'results_suffix')
+    output_dir = os.path.join(args['workspace_dir'], 'output')
+    intermediate_dir = os.path.join(args['workspace_dir'], 'intermediate')
+    taskgraph_cache_dir = os.path.join(intermediate_dir, 'task_cache')
 
+    utils.make_directories([output_dir, intermediate_dir, taskgraph_cache_dir])
 
-def s_to_timestep(snapshot_years, snapshot_idx):
-    """Convert snapshot index position to timestep.
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # Synchronous mode.
+    task_graph = taskgraph.TaskGraph(
+        taskgraph_cache_dir, n_workers, reporting_interval=5.0)
+
+    snapshots = _extract_snapshots_from_table(args['landcover_snapshot_csv'])
+
+    # Phase 1: alignment and preparation of inputs
+    baseline_lulc_year = min(snapshots.keys())
+    baseline_lulc_path = snapshots[baseline_lulc_year]
+    baseline_lulc_info = pygeoprocessing.get_raster_info(
+        baseline_lulc_path)
+    min_pixel_size = numpy.min(numpy.abs(baseline_lulc_info['pixel_size']))
+    target_pixel_size = (min_pixel_size, -min_pixel_size)
```

Comment:
I believe it will be ok, and you do make a really interesting point here.  I've made a note to include this in the updated UG, just to clarify the behavior for anyone who's wondering.

Interestingly, the model's former implementation actually did two things: the preprocessor used the pixel minimum, and the model resampled to the baseline.  So in a way, using the minimum resolution would be adhering to (part of) the original behavior.  In this case, because we're working with discrete/classified values, I suspect that the underlying results will be OK given that there's already a degree of error stemming from discretization.

In practice, all of the CBC model runs I've ever seen have all had very similar or identical cell sizes.